dlib/test/dnn.cpp

```diff
@@ -2002,6 +2002,18 @@ namespace
             auto res = test_layer(l);
             DLIB_TEST_MSG(res, res);
         }
+        {
+            print_spinner();
+            clipped_relu_ l;
+            auto res = test_layer(l);
+            DLIB_TEST_MSG(res, res);
+        }
+        {
+            print_spinner();
+            elu_ l;
+            auto res = test_layer(l);
+            DLIB_TEST_MSG(res, res);
```

Comment:
I don't know why ELU is failing... I've run some tests and the CPU version works well, but not the CUDA one, which is surprising, because I'm just calling the CUDNN implementation...